commerce-service/src/main/java/com/liferay/commerce/internal/messaging/CommerceOrderStatusMessageListener.java

```diff
@@ -44,27 +46,89 @@
 	protected void doReceive(Message message) throws Exception {
 		int orderStatus = message.getInteger("orderStatus");
 
-		if (orderStatus == CommerceOrderConstants.ORDER_STATUS_TO_TRANSMIT) {
-			long commerceOrderId = message.getLong("commerceOrderId");
+		long commerceOrderId = message.getLong("commerceOrderId");
 
-			CommerceOrder commerceOrder =
-				_commerceOrderLocalService.getCommerceOrder(commerceOrderId);
+		CommerceOrder commerceOrder =
+			_commerceOrderLocalService.getCommerceOrder(commerceOrderId);
 
-			// Commerce notifications
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.getCommerceChannelByOrderGroupId(
+				commerceOrder.getGroupId());
 
-			CommerceChannel commerceChannel =
-				_commerceChannelLocalService.getCommerceChannelByOrderGroupId(
-					commerceOrder.getGroupId());
+		boolean hasWorkflow =
+			_workflowDefinitionLinkLocalService.hasWorkflowDefinitionLink(
+				commerceOrder.getCompanyId(), commerceOrder.getGroupId(),
+				CommerceOrder.class.getName(), 0,
+				CommerceOrderConstants.TYPE_PK_APPROVAL);
+
+		if (hasWorkflow &&
+			(orderStatus == CommerceOrderConstants.ORDER_STATUS_OPEN) &&
+			(commerceOrder.getStatus() != WorkflowConstants.STATUS_APPROVED)) {
 
 			_commerceNotificationHelper.sendNotifications(
 				commerceChannel.getSiteGroupId(),
-				OrderPlacedCommerceNotificationTypeImpl.KEY, commerceOrder);
+				CommerceOrderStatusNotificationConstants.ORDER_SUBMITTED,
+				commerceOrder);
+		}
+		else if (hasWorkflow &&
+				 (orderStatus == CommerceOrderConstants.ORDER_STATUS_OPEN) &&
+				 (commerceOrder.getStatus() ==
```

Comment:
This conditional test `(commerceOrder.getStatus() == WorkflowConstants.STATUS_APPROVED)` is technically redundant -- we should probably remove it